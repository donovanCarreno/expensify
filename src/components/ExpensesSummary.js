import React from 'react'
import { connect } from 'react-redux'
import numeral from 'numeral'
import expensesTotal from '../selectors/expenses-total'
import selectExpenses from '../selectors/expenses'

// export for testing
export const ExpensesSummary = ({ total, count }) => (
  <div>
    Viewing {count === 1 ? `1 expense totaling ${numeral(total / 100).format('$0,0.00')}` : `${count} expenses totaling ${numeral(total / 100).format('$0,0.00')}`}
  </div>
)

const mapStateToProps = (state) => ({
  total: expensesTotal(selectExpenses(state.expenses, state.filters)),
  count: selectExpenses(state.expenses, state.filters).length
})

export default connect(mapStateToProps)(ExpensesSummary)
