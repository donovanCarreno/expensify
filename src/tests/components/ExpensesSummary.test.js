import React from 'react'
import { shallow } from 'enzyme'
import { ExpensesSummary } from '../../components/ExpensesSummary'
import expenses from '../fixtures/expenses'
import totalExpenses from '../../selectors/expenses-total'

test('should render ExpensesSummary with 1 expense', () => {
  const count = 1
  const total = totalExpenses(expenses[0])
  const wrapper = shallow(<ExpensesSummary count={count} total={total} />)
  expect(wrapper).toMatchSnapshot()
})

test('should render ExpensesSummary with 2 expenses', () => {
  const count = 2
  const total = totalExpenses([expenses[0], expenses[1]])
  const wrapper = shallow(<ExpensesSummary count={count} total={total} />)
  expect(wrapper).toMatchSnapshot()
})
