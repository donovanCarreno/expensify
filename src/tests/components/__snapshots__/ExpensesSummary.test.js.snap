// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary with 1 expense 1`] = `
<div>
  Viewing 
  1 expense totaling $1.95
</div>
`;

exports[`should render ExpensesSummary with 2 expenses 1`] = `
<div>
  Viewing 
  2 expenses totaling $196.95
</div>
`;
